lib/internal/modules/helpers.js

```diff
@@ -288,14 +289,30 @@ function addBuiltinLibsToObject(object, dummyModuleName) {
 }
 
 /**
- * If a referrer is an URL instance or absolute path, convert it into an URL string.
- * @param {string | URL} referrer
+ * Normalize the referrer name as a URL.
+ * If it's an absolute path or a file:// it's normalized as a file:// URL.
+ * Otherwise it's returned as undefined;
+ * @param {string | null | undefined | false | any } referrerName
+ * @returns {string | undefined}
  */
-function normalizeReferrerURL(referrer) {
-  if (typeof referrer === 'string' && path.isAbsolute(referrer)) {
-    return pathToFileURL(referrer).href;
+function normalizeReferrerURL(referrerName) {
+  if (typeof referrerName !== 'string') {
+    return undefined;
```

Comment:
It's a bit weird that `URL` instances are not accepted if this value is user provided.

Reply:
They are not going to be URL, or any objects in that matter, because only primitives are allowed as resource names. We get this value from V8.

Reply:
Added an internal assertion that it can never be anything but null, undefined, or a string.